pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1408,15 +1408,25 @@ func (l *LibvirtDomainManager) allocateHotplugPorts(
 }
 
 func getSourceFile(disk api.Disk) string {
-	source := disk.Source
-	if source.DataStore != nil {
-		source = *source.DataStore.Source
+	if disk.Source.File != "" {
+		return disk.Source.File
 	}
-	file := source.File
-	if source.File == "" {
-		file = disk.Source.Dev
+
+	if disk.Source.Dev != "" {
+		return disk.Source.Dev
+	}
+	return ""
+}
+
+func getBackendSource(disk api.Disk) string {
+	if disk.Source.DataStore != nil && disk.Source.DataStore.Source != nil {
```

Comment:
is it even possible to have disk.Source.DataStore != nil && disk.Source.DataStore.Source == nil ?